Makefile

```
@@ -1044,6 +1044,28 @@ grpc:
 grpc/host: protos/all
 	@build.assets/genproto.sh
 
+# protos-up-to-date checks if the generated GRPC stubs are up to date.
+# This target runs in the buildbox container.
+.PHONY: protos-up-to-date
+protos-up-to-date:
+	$(MAKE) -C build.assets protos-up-to-date
+
+# protos-up-to-date/host checks if the generated GRPC stubs are up to date.
+# Unlike protos-up-to-date, this target runs locally.
+.PHONY: protos-up-to-date/host
+protos-up-to-date/host: must-start-clean/host grpc/host
+	@if ! build.assets/is-repo-clean.sh; then \
+		echo 'Please run make grpc.'; \
+		exit 1; \
+	fi
```

Why not?

```suggestion
	@if ! $(GIT) diff --quiet; then \
		echo 'Please run make grpc.'; \
		exit 1; \
	fi
```

and then define at the top:

```Makefile
GIT ?= git
```

`git diff --quiet` is a relatively simple, straightforward command, I don't see much harm in repeating it.

---

I removed it and then added it back in just due to @rosstimothy's comment that it could be useful for other purposes, but I'm fine going either way.

---

After talking with Tim, I'm going to take your suggestions and remove the helper script.